static/js/libs/channels.js

```diff
@@ -0,0 +1,262 @@
+const RISKS = ["stable", "candidate", "beta", "edge"];
+
+/**
+ * Parse a channel string into an object
+ * @param {string} channelString
+ * @param {{defaultTrack: string}} options
+ * @returns {{format: {risk: boolean, track: boolean, branch: boolean}, risk: *, track: (string|*), branch: (string|*)}}
+ */
+function parseChannel(channelString, options) {
+  const format = {
+    track: false,
+    risk: false,
+    branch: false
+  };
+  const channelArr = ["latest", undefined, "_base"];
```

Comment:
Is there any reason for doing this in array instead of 3 variables (`track`, `risk`, `branch`)?

Also isn't the default branch empty? If we start using "_base" we may end up detecting a branch when there isn't any. But maybe as we approach adding support for branches in more elements of UI we should just start using "_base" as default. If that's the case it would be good to have "_base" (but "latest" as well) defined somewhere as consts and reused instead of being copied.

Reply:
So I introduced the "_base" as a concept to have both a common format for all track/risks, regardless of whether they have branches and also to help with parsing in other parts in this file, for example reducing/ sorting `["stable/test", "stable"]` was producing `["stable/test"]` so stripping the "base" stable channel.

Setting "_base" hopefully makes it clear for anyone calling this code in the future (for another part of the site)